server/graphql/asset/src/types/asset.rs

```diff
@@ -0,0 +1,342 @@
+        &self.row().created_datetime
+    }
+
+    pub async fn modified_datetime(&self) -> &chrono::NaiveDateTime {
+        &self.row().modified_datetime
+    }
+
+    pub async fn store(&self, ctx: &Context<'_>) -> Result<Option<StoreNode>> {
+        let store_id = match &self.row().store_id {
+            Some(store_id) => store_id,
+            None => return Ok(None),
+        };
+
+        let loader = ctx.get_loader::<DataLoader<StoreByIdLoader>>();
+        Ok(loader
+            .load_one(store_id.clone())
+            .await?
+            .map(StoreNode::from_domain))
+    }
+
+    pub async fn catalogue_item(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Option<AssetCatalogueItemNode>> {
+        let catalogue_item_id = match &self.row().catalogue_item_id {
+            Some(catalogue_item_id) => catalogue_item_id,
+            None => return Ok(None),
+        };
+
+        let loader = ctx.get_loader::<DataLoader<AssetCatalogueItemLoader>>();
+        Ok(loader
+            .load_one(catalogue_item_id.clone())
+            .await?
+            .map(AssetCatalogueItemNode::from_domain))
+    }
+
+    pub async fn locations(&self, ctx: &Context<'_>) -> Result<LocationConnector> {
+        let asset_id = &self.row().id;
+        let loader = ctx.get_loader::<DataLoader<AssetLocationLoader>>();
+        let result_option = loader.load_one(asset_id.to_string()).await?;
+
+        let locations = LocationConnector::from_vec(result_option.unwrap_or(vec![]));
+
+        Ok(locations)
+    }
+
+    pub async fn documents(&self, ctx: &Context<'_>) -> Result<SyncFileReferenceConnector> {
+        let asset_id = &self.row().id;
+        let loader = ctx.get_loader::<DataLoader<SyncFileReferenceLoader>>();
+        let result_option = loader.load_one(asset_id.to_string()).await?;
+
+        let documents = SyncFileReferenceConnector::from_vec(result_option.unwrap_or(vec![]));
+
+        Ok(documents)
+    }
+
+    // pub async fn properties(
+    //     &self,
+    //     ctx: &Context<'_>,
+    // ) -> Result<Vec<AssetCatalogueItemPropertyValueNode>> {
+    //     let properties = match &self.row().catalogue_item_id {
+    //         Some(catalogue_item_id) => {
+    //             let loader = ctx.get_loader::<DataLoader<AssetCatalogueItemPropertyLoader>>();
+    //             let result_option = loader.load_one(catalogue_item_id.to_string()).await?;
+
+    //             result_option
+    //                 .unwrap_or(Vec::<AssetCatalogueItemPropertyValue>::new())
+    //                 .iter()
+    //                 .map(|p| AssetCatalogueItemPropertyValueNode::from_domain(p.to_owned()))
+    //                 .into_iter()
+    //                 .collect()
+    //         }
+    //         None => vec![],
+    //     };
+
+    //     Ok(properties)
+    // }
+
+    pub async fn properties(&self) -> Result<String> {
+        // TODO: Merge properties with catalogue item properties
```

Comment:
Obviously this looses all the asset catalogue item properties.
Was going to merge here, but wonder if we should return both as different fields and merge in the frontend.
That would also let us show if something's been overridden for example?